tests/search/search_test.php

```diff
@@ -0,0 +1,234 @@
+
+        $record = new \stdClass();
+        $record->course = $course->id;
+        $record->customfield_overview = 'Test overview text 2';
+        $record->typeid = $this->cmstype->get('id');
+        $generator->create_instance_with_data($record);
+
+        // All records.
+        $recordset = $searcharea->get_document_recordset();
+        $this->assertTrue($recordset->valid());
+        $this->assertEquals(2, iterator_count($recordset));
+        $recordset->close();
+
+        // The +2 is to prevent race conditions.
+        $recordset = $searcharea->get_document_recordset(time() + 2);
+
+        // No new records.
+        $this->assertFalse($recordset->valid());
+        $recordset->close();
+
+        // Wait 1 sec to have new search string.
+        sleep(1);
+        $record = new \stdClass();
+        $record->course = $course->id;
+        $record->customfield_overview = 'Test overview text 3';
+        $record->typeid = $this->cmstype->get('id');
+        $generator->create_instance_with_data($record);
+
+        // Return only new search.
+        $recordset = $searcharea->get_document_recordset(time());
+        $count = 0;
+        foreach ($recordset as $record) {
+            $this->assertInstanceOf('stdClass', $record);
+            $data = $DB->get_record('customfield_data', ['id' => $record->id]);
+            $doc = $searcharea->get_document($record);
+            $this->assertInstanceOf('\core_search\document', $doc);
+            $this->assertEquals('mod_cms-cmsfield-' . $data->id, $doc->get('id'));
+            $this->assertEquals($data->id, $doc->get('itemid'));
+            $this->assertEquals($course->id, $doc->get('courseid'));
+            $this->assertEquals($data->contextid, $doc->get('contextid'));
+            $this->assertEquals($this->field->get('name'), $doc->get('title'));
+            $this->assertEquals($data->value, $doc->get('content'));
+
+            // Static caches are working.
+            $dbreads = $DB->perf_get_reads();
+            $doc = $searcharea->get_document($record);
+            $this->assertEquals($dbreads, $DB->perf_get_reads());
+            $this->assertInstanceOf('\core_search\document', $doc);
+            $count++;
+        }
+        $this->assertEquals(1, $count);
+        $recordset->close();
+    }
+
+    /**
+     * Document contents.
+     *
+     * @return void
+     */
+    public function test_check_access(): void {
+        global $DB;
+
+        // Returns the instance as long as the area is supported.
+        $searcharea = \core_search\manager::get_search_area($this->cmsareaid);
+
+        $user1 = self::getDataGenerator()->create_user();
+        $user2 = self::getDataGenerator()->create_user();
+        $course = self::getDataGenerator()->create_course();
+
+        $this->getDataGenerator()->enrol_user($user1->id, $course->id, 'student');
+
+        // Name for cms is coming from "title_mustache" in cms_type.
```

Comment:
What does this comment mean? How is it relevant to the code?

Reply:
I updated the comment.
Basically, any activity should have "name".
But for cms activity, the name is coming from "title_mustache" in cms_type table.
I was confused about the name, so I added the comment.